torch/_inductor/scheduler.py

```diff
@@ -76,6 +83,12 @@
 fusion_log = torch._logging.getArtifactLogger(__name__, "fusion")
 loop_ordering_log = torch._logging.getArtifactLogger(__name__, "loop_ordering")
 
+PartitionType = List["BaseSchedulerNode"]
+
+# Mapping from partition input names to IRNode/Expr, and a boolean for whether
+# deallocating it after the partition
+PartitionInputMetadataType = Dict[str, Tuple[Union[ir.IRNode, sympy.Expr], bool]]
```

Comment:
`Tuple[Union[ir.IRNode, sympy.Expr], bool]` -  i would lean towards using a data class for this.